### PR TITLE
Setup proto gen for denyChecker config.

### PR DESCRIPTION
**Pull request by douglas-reid**
This should address #125 and #127.

@geeknoid please let me know if this is a reasonable direction.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/130)

### CODE REVIEW

**Comment by geeknoid** [comment]
This definitely works. Mandar was trying to get things such that the Config type was actually in the same package as the rest of the adapter code. I'm OK with this organization though if he is. And I definitely like not having the pb.go file checked in.

**Comment by mandarjog** [comment]
did you try using the same package as the code ?
The only issue I had with it was co-existence with go tools.

File: adapter/denyChecker/adapter_test.go [hunk 1]
```diff
@@ -28,23 +31,23 @@ func TestAll(t *testing.T) {
...
+	a, err = b.NewAspect(&pb.Config{&status.Status{Code: int32(code.Code_INVALID_ARGUMENT)}})
 	if err != nil {
 		t.Errorf("Unable to create aspect: %v", err)
 	}
 
-	status = a.Deny()
-	if status.Code != int32(code.Code_INVALID_ARGUMENT) {
-		t.Errorf("a.Deny returned %d, expected %d", status.Code, int32(code.Code_INVALID_ARGUMENT))
+	s = a.Deny()
+	if s.Code != int32(code.Code_INVALID_ARGUMENT) {
```
Comment (douglas-reid) [inline adapter/denyChecker/adapter_test.go]: code.Code is an enum. status.Code is an int32. without the cast, you will get something like: `invalid operation: ... (mismatched types int32 and Code)`

**Comment by douglas-reid** [comment]
I did not try to merge packages. I wasn't aware this was a goal. I'm not convinced that is the right thing to do, but I can play with it to see how such a thing _might_ work. I'm skeptical that it won't cause issues and/or confusion.

**Comment by mandarjog** [comment]
I think being able to keep all adapter code together is a good goal, but visually it is together since the config file is right there. As a new adapter author I would find it confusing as to why is this not in the current directory.  

That said, it is ok for now. We can revisit it later.

**Pull request merged**